src/clients/SpokePoolClient.ts

```diff
@@ -239,6 +257,109 @@ export class SpokePoolClient {
     return `${event.depositId}-${event.originChainId}`;
   }
 
+  // Look for the block number of the event that emitted the deposit with the target deposit ID. We know that
+  // `numberOfDeposits` is strictly increasing for any SpokePool, so we can use a binary search to find the blockTag
+  // where `numberOfDeposits == targetDepositId`.
+  async binarySearchForBlockContainingDepositId(
+    targetDepositId: number,
+    fallback: BINARY_SEARCH_FALLBACK = "NONE"
+  ): Promise<number> {
+    // After an update(), firstBlockToSearch increments to the toBlock+1 so clamp that at the latestBlockNumber to
+    // avoid calling spokePool.numberOfDeposits for a block that doesn't exist yet.
+    let high = this.latestBlockNumber
+      ? Math.min(this.firstBlockToSearch, this.latestBlockNumber)
+      : this.firstBlockToSearch;
```

Comment:
Potential simplification:

```suggestion
    let high = Math.min(this.firstBlockToSearch, this.latestBlockNumber ?? this.firstBlockToSearch);
```